program/c/src/oracle/upd_aggregate.h

```diff
@@ -18,11 +18,10 @@ typedef struct pc_qset
   pd_t      weight_[PC_COMP_SIZE];
   int64_t   decay_[1+PC_MAX_SEND_LATENCY];
   int64_t   fact_[PC_FACTOR_SIZE];
-  int32_t   expo_;
 } pc_qset_t;
 
 // initialize quote-set temporary data in heap area
-static pc_qset_t *qset_new( int expo )
+static pc_qset_t *qset_new()
```

Comment:
This change here fixes the nondeterministic test. Turns out that pc_qset_t is shared across threads and constant *except* it stores a copy of the exponent, making it unsafe to invoke this method in parallel (which`cargo test` does by default). There's no reason it needs to store the exponent -- it's just bad software design.

fix is easy: just make it constant so it's threadsafe.